app-libs/stf/src/getter.rs

```diff
@@ -0,0 +1,198 @@
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{AccountId, KeyPair, Signature};
+use codec::{Decode, Encode};
+use itp_stf_interface::ExecuteGetter;
+use log::*;
+use sp_runtime::traits::Verify;
+use std::prelude::v1::*;
+
+#[cfg(feature = "evm")]
+use sp_core::{H160, H256};
+
+#[cfg(feature = "sgx")]
+pub use sgx::*;
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+#[allow(non_camel_case_types)]
+pub enum Getter {
+	public(PublicGetter),
+	trusted(TrustedGetterSigned),
+}
+
+impl From<PublicGetter> for Getter {
+	fn from(item: PublicGetter) -> Self {
+		Getter::public(item)
+	}
+}
+
+impl From<TrustedGetterSigned> for Getter {
+	fn from(item: TrustedGetterSigned) -> Self {
+		Getter::trusted(item)
+	}
+}
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+#[allow(non_camel_case_types)]
+pub enum PublicGetter {
+	some_value,
+}
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+#[allow(non_camel_case_types)]
+pub enum TrustedGetter {
+	free_balance(AccountId),
+	reserved_balance(AccountId),
+	nonce(AccountId),
+	#[cfg(feature = "evm")]
+	evm_nonce(AccountId),
+	#[cfg(feature = "evm")]
+	evm_account_codes(AccountId, H160),
+	#[cfg(feature = "evm")]
+	evm_account_storages(AccountId, H160, H256),
+}
+
+impl TrustedGetter {
+	pub fn sender_account(&self) -> &AccountId {
+		match self {
+			TrustedGetter::free_balance(sender_account) => sender_account,
+			TrustedGetter::reserved_balance(sender_account) => sender_account,
+			TrustedGetter::nonce(sender_account) => sender_account,
+			#[cfg(feature = "evm")]
+			TrustedGetter::evm_nonce(sender_account) => sender_account,
+			#[cfg(feature = "evm")]
+			TrustedGetter::evm_account_codes(sender_account, _) => sender_account,
+			#[cfg(feature = "evm")]
+			TrustedGetter::evm_account_storages(sender_account, ..) => sender_account,
+		}
+	}
+
+	pub fn sign(&self, pair: &KeyPair) -> TrustedGetterSigned {
+		let signature = pair.sign(self.encode().as_slice());
+		TrustedGetterSigned { getter: self.clone(), signature }
+	}
+}
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub struct TrustedGetterSigned {
+	pub getter: TrustedGetter,
+	pub signature: Signature,
+}
+
+impl TrustedGetterSigned {
+	pub fn new(getter: TrustedGetter, signature: Signature) -> Self {
+		TrustedGetterSigned { getter, signature }
+	}
+
+	pub fn verify_signature(&self) -> bool {
+		self.signature
+			.verify(self.getter.encode().as_slice(), self.getter.sender_account())
+	}
+}
+
+#[cfg(feature = "sgx")]
+pub mod sgx {
+	use super::*;
+	use ita_sgx_runtime::System;
+	use itp_utils::stringify::account_id_to_string;
+
+	#[cfg(feature = "evm")]
+	use ita_sgx_runtime::{AddressMapping, HashedAddressMapping};
+
+	#[cfg(feature = "evm")]
+	use crate::evm_helpers::{get_evm_account, get_evm_account_codes, get_evm_account_storages};
+
+	impl ExecuteGetter for Getter {
+		fn execute(self) -> Option<Vec<u8>> {
```

Comment:
No logic change, but I did the following:
- move getter defintion from `lib.rs` to separate file
- getter no implements their execution themselves.